pkg/controllers/provisioning/controller.go

```diff
@@ -20,42 +20,44 @@ import (
 	"context"
 	"time"
 
+	"github.com/samber/lo"
 	v1 "k8s.io/api/core/v1"
 	controllerruntime "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
 	"sigs.k8s.io/karpenter/pkg/events"
 	operatorcontroller "sigs.k8s.io/karpenter/pkg/operator/controller"
 	"sigs.k8s.io/karpenter/pkg/utils/pod"
 )
 
-var _ operatorcontroller.TypedController[*v1.Pod] = (*Controller)(nil)
+var _ operatorcontroller.TypedController[*v1.Pod] = (*PodController)(nil)
 
 // Controller for the resource
-type Controller struct {
+type PodController struct {
 	kubeClient  client.Client
 	provisioner *Provisioner
 	recorder    events.Recorder
 }
 
 // NewController constructs a controller instance
-func NewController(kubeClient client.Client, provisioner *Provisioner, recorder events.Recorder) operatorcontroller.Controller {
-	return operatorcontroller.Typed[*v1.Pod](kubeClient, &Controller{
+func NewPodController(kubeClient client.Client, provisioner *Provisioner, recorder events.Recorder) operatorcontroller.Controller {
+	return operatorcontroller.Typed[*v1.Pod](kubeClient, &PodController{
 		kubeClient:  kubeClient,
 		provisioner: provisioner,
 		recorder:    recorder,
 	})
 }
 
-func (c *Controller) Name() string {
-	return "provisioner.trigger"
+func (*PodController) Name() string {
+	return "provisioner.podtrigger"
```

Comment:
musing: Is this `provisioner.podtrigger` or `provisioner.trigger.pod` with the directory structure modeling that too?